src/app/app.component.ts

```diff
@@ -8,22 +8,21 @@ import { Component } from '@angular/core';
 export class AppComponent {
   title = 'VideoApp';
   newTask: string;
-  tasksList: Array<string> = [];
+  films: Array<string> = [];
   tasksFavorite: Array<string> = [];
 
+
   add() {
-    this.tasksList.push(this.newTask);
+    this.films.push(this.newTask);
```

Comment:
Póki masz jeden komponent w apce to jest jeszcze spoko, ale gdybyś miał kilka komponentów i wszystkie by były zainteresowane dostępem do filmów, a oprócz filmów miałbyś jeszcze inne dane, to nagle byłby straszny burdel w tym App.Component gdzie przechowywane byłyby wszystkie dane.

Takim najbardziej dojrzaym rozwiązaniem jest stosowanie reduxa i globalnego store, ale może nie warto się tutaj w to zapuszczać. Innym akceptowalnym rozwiązaniem jest zrobienie serwisów do trzymania stanu i podpięcie ich np. w core.module, ew w app.module (ważne żeby były wpięte jeden raz w aplikacji). Także polecam Ci zrobić films.service, na którym będzie tablica films i metody getFilms,addFilm, removeFilm które będą mogły wywoływac inne części aplikacji. W app.component sobie zaimportujesz ten serwis w konstruktorze i będziesz używał this.filmsService.addFilm(this.newFilm)

Reply:
Druga uwaga: jakoś bardziej semantycznie by było jakbyś to napisał add(film: Film):void { this.flms.push(film) }; wiem, że zrobiłeś to 2 way data binding do inputu, ale byłoby lepiej jakbyś do funkcji dodającej film podawał film jako argument, a nie brał go gdzies z dupy spoza funkcji. Generalnie jest to akceptowalne, zwłaszcza na początku, więc jak nie rozkminisz tego z 30minut to daj spokój na razie

Reply:
właśnie ten burdel z taskList/films mam zamiar dzisiaj ogarnąć :/